### DISCUSSION TITLE
PCA/UMAP endpoints

### DISCUSSION THREAD

**Discussion by janash** [Descriptor-Libraries/descriptor-libraries-framework#8] [Ideas]
What does everyone think of having an endpoint that is:

`api/molecules/dimensions`

where we have query parameters

type (str, umap or pca)
components (list of numbers separated by commas, same as neighbors function where defaults are set based on type)
skip (optional, int)
limit (optional, int)

So calling the API would look like `api/molecules/components?type=pca&limit=100` (with some parameters as an example)

We could add category (optional, string) as well to make it more similar to the current umap endpoint. Without a category, it would return all categories.

The should use a similar schema to `MoleculeNeighbors` (without distance). I wonder if we could use inheritance here:

```
class MoleculeComponets(BaseModel):
    type: Optional[str]
    molecule_id: int
    pat: Optional[str]
    smiles: str
    components: list[float]

class MoleculeNeighbors(MoleculeComponents):
        dist: Optional[float]
```

@ujamshed

**Comment by ujamshed** [comment]
Created in PR #9.

**Reply by ujamshed** [reply]
@janash Should the users be able to pass in a number of categories they want to return? Like pc3, pn3, po3? or are we sticking to singular categories at this time.
